src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountCertificates.ps1

```diff
@@ -0,0 +1,95 @@
+    [parameter(Mandatory = $false)][string] $CertificatesFolder = $(if ($CertificateFilePath -eq '') { throw "Either the file path of a specific certificate or the file path of a folder containing multiple certificates is required, e.g.: -CertificateFilePath 'C:\Certificates\certificate.cer' or -CertificatesFolder 'C:\Certificates'" }),
+    [Parameter(Mandatory = $false)][string] $KeyName = $(if ($CertificateType -eq 'Private') { throw "If the CertificateType is set to 'Private', the KeyName must be supplied" }),
+    [Parameter(Mandatory = $false)][string] $KeyVersion = $(if ($CertificateType -eq 'Private') { throw "If the CertificateType is set to 'Private', the KeyVersion must be supplied" }),
+    [Parameter(Mandatory = $false)][string] $KeyVaultId = $(if ($CertificateType -eq 'Private') { throw "If the CertificateType is set to 'Private', the KeyVaultId must be supplied" }),
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($CertificateFilePath -ne '' -and $CertificatesFolder -ne '') {
+    throw "Either the file path of a specific certificate or the file path of a folder containing multiple certificates is required, e.g.: -CertificateFilePath 'C:\Certificates\certificate.cer' or -CertificatesFolder 'C:\Certificates'"
+}
+
+if ($CertificateType -ne 'Public' -and $CertificateType -ne 'Private') {
+    throw "The CertificateType should be either 'Public' or 'Private'"
+}
+
+if ($CertificateType -eq 'Private' -and $CertificatesFolder -ne '' -and $CertificateFilePath -eq '') {
+    throw "Using the CertificatesFolder parameter in combination with Private certificates is not possible, since this would upload multiple certificates using the same Key in Azure KeyVault"
+}
+
+function UploadCertificate {
+    param
+    (
+        [System.IO.FileInfo][parameter(Mandatory = $true)]$Certificate
+    )
+
+    $certificateName = $Certificate.BaseName
+    if ($ArtifactsPrefix -ne '') {
+        $certificateName = $ArtifactsPrefix + $certificateName
+    }
+    Write-Host "Uploading certificate '$certificateName' into the Integration Account '$Name'"
+
+    $existingCertificate = $null
+    try {
+        Write-Verbose "Checking if the certificate '$certificateName' already exists in the Integration Account '$Name'"
+        $existingCertificate = Get-AzIntegrationAccountCertificate -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -CertificateName $certificateName -ErrorAction Stop
+    }
+    catch {
+        if ($_.Exception.Message.Contains('could not be found')) {
+            Write-Verbose "No certificate '$certificateName' could not be found in Azure Integration Account '$Name'"
+        }
+        else {
+            throw $_.Exception
+        }
+    }
```

Comment:
For my information: in what scenario are we re-throwing the exception? This is (I think) purely to bubble up any problems with the Azure Integration Account.

Reply:
Yes, the rethrow makes sure that the script does not continue with uploading the files as something other than `could not be found` has gone wrong while check if the certificate already exists.

Reply:
The rethrow will be caught and written to the log with `Write-Error "Failed to upload certificate '$certificateName' in Azure Integration Account '$Name': '$($_.Exception.Message)'"` at line 75